modules/gapi/CMakeLists.txt

```diff
@@ -178,12 +181,18 @@ ocv_module_include_directories("${CMAKE_CURRENT_LIST_DIR}/src")
 ocv_create_module()
 
 ocv_target_link_libraries(${the_module} PRIVATE ade)
+
 if(OPENCV_GAPI_INF_ENGINE)
   ocv_target_link_libraries(${the_module} PRIVATE ${INF_ENGINE_TARGET})
 endif()
 if(HAVE_TBB)
   ocv_target_link_libraries(${the_module} PRIVATE tbb)
 endif()
+if(CV_TRACE AND HAVE_ITT)
+  add_definitions(-DOPENCV_WITH_ITT=1)
```

Comment:
> `add_definitions(-DOPENCV_WITH_ITT=1)`

Again, why do you it globally? Shouldn't it be a target-specific definition via `target_compile_definitions`?

Reply:
AND maybe OpenCV has already its own module-related macro for it.

Reply:
What do you mean under "module-related" macro? It seems that ITT is used in core module as a backend for OpenCV traces API and has the only "OPENCV_WITH_ITT" define..
@alalek , Alexander, could you please prove or correct this information?

Reply:
target_compile_definitions() should be better (CMake 3.5.1 has good support for that)

`OPENCV_WITH_ITT` is used with "core" module only (for now).